### DISCUSSION TITLE
which one is better for saving messages in a chat app, rabbitmq rpc or rabbitmq work(task) queues?

### DISCUSSION THREAD

**Discussion by hosmoh** [rabbitmq/rabbitmq-server#2782] [Other] [answered]
I have a chat application that is written with node.js and consists of two micro-services: one for connecting to the database and storing messages and the other which is for chat and uses socket.io.

These two micro-services have been deployed on the server using docker.

When a user sends a message, I immediately broadcast that message in the chat and then use RabbitMQ to save it to the database.

Am i doing this right and if the answer is yes, which is more suitable for me between RabbitMQ RPC and RabbitMQ Work Queues ?

**Comment by michaelklishin** [comment]
I will convert this issue to a GitHub discussion. Currently GitHub will [automatically close and lock the issue](https://github.com/github/feedback/discussions/3197) even though your question will be transferred and responded to elsewhere. This is to let you know that we do not intend to ignore this but this is how the current GitHub conversion mechanism makes it seem for the users :(

**Comment by michaelklishin** [comment, marked as answer]
Request/reply ("RPC") assumes that a client (a component of your system) requires an asynchronous response. The pattern from tutorial two assumes no response. For archiving purposes specifically, the latter sounds sufficient to me but I know nothing about your system, of course.

**Reply by hosmoh** [reply]
I greatly appreciate your consideration
